knative-operator/pkg/common/service_monitor.go

```diff
@@ -12,97 +10,76 @@ import (
-	SourceNameLabel                  = "eventing.knative.dev/sourceName"
-	SourceRoleLabel                  = "sources.knative.dev/role"
-	TestMonitor                      = "TEST_MONITOR"
-	TestSourceServiceMonitorPath     = "TEST_SOURCE_SERVICE_MONITOR_PATH"
-	TestSourceServicePath            = "TEST_SOURCE_SERVICE_PATH"
+	SourceLabel     = "eventing.knative.dev/source"
+	SourceNameLabel = "eventing.knative.dev/sourceName"
+	SourceRoleLabel = "sources.knative.dev/role"
 )
 
-func SetupSourceServiceMonitor(client client.Client, instance *appsv1.Deployment) error {
+func SetupSourceServiceMonitorResources(client client.Client, instance *appsv1.Deployment) error {
 	labels := instance.Spec.Selector.MatchLabels
-
 	clientOptions := mf.UseClient(mfclient.NewClient(client))
-	// create service for the deployment
-	manifest, err := mf.NewManifest(getMonitorPath(TestSourceServicePath, EventingSourcePath), clientOptions)
+	// Create service monitor resources for source
+	smManifest, err := createServiceMonitorManifest(labels, instance.Name, instance.Namespace, clientOptions)
 	if err != nil {
-		return fmt.Errorf("unable to parse source service manifest: %w", err)
-	}
-	transforms := []mf.Transformer{updateService(labels, instance.Name), mf.InjectOwner(instance), mf.InjectNamespace(instance.Namespace)}
-	if manifest, err = manifest.Transform(transforms...); err != nil {
-		return fmt.Errorf("unable to transform source service manifest: %w", err)
-	}
-	if err := manifest.Apply(); err != nil {
 		return err
 	}
-
-	// get service back, needed for the UID and setting owner refs
-	srv := &v1.Service{}
-	if err := client.Get(context.TODO(), types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}, srv); err != nil {
-		return err
-	}
-	// create service monitor for source
-	manifest, err = mf.NewManifest(getMonitorPath(TestSourceServiceMonitorPath, EventingSourceServiceMonitorPath), clientOptions)
-	if err != nil {
-		return fmt.Errorf("unable to parse source service monitor manifest: %w", err)
-	}
-	transforms = []mf.Transformer{updateServiceMonitor(labels, instance.Name), mf.InjectOwner(srv), mf.InjectNamespace(instance.Namespace)}
-	if manifest, err = manifest.Transform(transforms...); err != nil {
+	if *smManifest, err = smManifest.Transform(mf.InjectOwner(instance)); err != nil {
 		return fmt.Errorf("unable to transform source service monitor manifest: %w", err)
 	}
-	return manifest.Apply()
+	return smManifest.Apply()
 }
 
-func getMonitorPath(envVar string, defaultVal string) string {
-	path := os.Getenv(envVar)
-	if path == "" {
-		return defaultVal
+func createServiceMonitorManifest(labels map[string]string, depName string, ns string, options mf.Option) (*mf.Manifest, error) {
+	var svU = &unstructured.Unstructured{}
+	var smU = &unstructured.Unstructured{}
+	sms := v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      depName,
+			Namespace: ns,
+			Labels:    kmeta.CopyMap(labels),
```

Comment:
Any particular reason we apply the same labels here that we match on? Same for the ServiceMonitor. Do we need labels at all (modulo the one to match on for the ServiceMonitor).

Reply:
These labels come from the deployment so I use them to tag the svc/sm too. It is a way to filter resources also from a cli perspective.

Reply:
They actually are the selector of the deployment though, right? So they'd be the same labels as on the pods.